### PR TITLE
drivers: esp32: i2c: configure clock speed during runtime

### PR DESCRIPTION
**Pull request by maass-hamburg**
- configure clock speed during runtime
- implement get_config of the i2c api

### CODE REVIEW

**Comment by maass-hamburg** [comment]
ping @sylvioalves

File: drivers/i2c/i2c_esp32.c [hunk 1]
```diff
@@ -262,13 +261,13 @@ static int i2c_esp32_recover(const struct device *dev)
 	return 0;
 }
 
-static void IRAM_ATTR i2c_esp32_configure_timeout(const struct device *dev)
+static void IRAM_ATTR i2c_esp32_configure_timeout(const struct device *dev, uint32_t bitrate)
```
Comment (sylvioalves) [inline drivers/i2c/i2c_esp32.c:264]: It sounds a bit strange to configure bit-rate in a `configure_timeout` function. I would either embed this into `i2c_esp32_configure()` or create a new `i2c_esp32_configure_btirate()` function. How does it sound? I guess 1st option is a bit straighforward.

Comment (maass-hamburg) [reply drivers/i2c/i2c_esp32.c]: fair point.
I now also moved the data mode part in a separate function, which is now only executed in the init, as it is set by the device tree and isn't changing during runtime.

**Comment by marekmatej** [comment]
@maass-hamburg can you please squash the commits into a single one?
Also, I think introducing the separate function `i2c_esp32_configure_bitrate` is better than hiding it under the timeout operation.

**Comment by maass-hamburg** [comment]
> @maass-hamburg can you please squash the commits into a single one? Also, I think introducing the separate function `i2c_esp32_configure_bitrate` is better than hiding it under the timeout operation.

@marekmatej is done.

**Comment by marekmatej** [comment]
> > @maass-hamburg can you please squash the commits into a single one? Also, I think introducing the separate function `i2c_esp32_configure_bitrate` is better than hiding it under the timeout operation.
> 
> @marekmatej is done.

well, renaming the `_timeout` function to `_bitrate` wasn't what I had on my mind :) But in the end, it is doing its job, so I'm fine with the changes.

**Review by marekmatej — CHANGES_REQUESTED** [review]
Could you please update the commit message, so it has a better description of the changes, and more importantly how to use this new feature?

**Review by marekmatej — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**